numerics/fma.hpp

```diff
@@ -0,0 +1,46 @@
+﻿#pragma once
+
+#include <cmath>
+#include <immintrin.h>
+
+#include "base/cpuid.hpp"
+#include "base/macros.hpp"
+
+namespace principia {
+namespace numerics {
+namespace internal_fma {
+
+using base::CPUFeatureFlags;
+
+// The functions in this file unconditionally wrap the appropriate intrinsics.
+// The caller may only use them if |UseHardwareFMA| is true.
+#if PRINCIPIA_USE_FMA_IF_AVAILABLE
+inline bool const UseHardwareFMA = HasCPUFeatures(CPUFeatureFlags::FMA);
+#else
+inline bool const UseHardwareFMA = false;
+#endif
+
+// ⟦ab + c⟧.
+inline double FusedMultiplyAdd(double a, double b, double c);
```

Comment:
Could we have tests for these functions to verify that we are calling the intrinsics correctly?